src/stories/Forms/InputWrapper/index.tsx

```diff
@@ -1,44 +1,147 @@
 import React from 'react';
 import styled from 'styled-components';
-import AlertText from '../../Foundation/AlertText';
 
-const Wrapper = styled.div`
+import { AlertText, Checkbox, Radio } from '../../../index';
+
+interface WrapperProps {
+  boxyErrorStyle?: boolean;
+  hasError?: boolean;
+}
+
+const Wrapper = styled.div<WrapperProps>`
   display: block;
+  width: 100%;
   max-width: 400px;
   margin-bottom: var(--spacing-32);
-`;
 
-const InputError = styled(AlertText)`
-  margin-top: var(--spacing-4);
+  .hint-wrapper {
+    position: relative;
+    margin-top: var(--spacing-4);
+    .hint {
+      &--help {
+        color: #747474;
+        font-size: var(--font-size-12);
+      }
+      // This is to make sure the hint/alert text doesn't take up space,
+      // as we don't want inputs moving around.
+      position: absolute;
+    }
+  }
+
+  // Apply special styling when boxyErrorStyle is true.
+  ${({ boxyErrorStyle }) =>
+    boxyErrorStyle &&
+    `.hint-wrapper .hint {
+      margin-left: var(--spacing-24);
+      &--error {
+        color: var(--color-alertRed);
+        position: static; // Make the error text take up space again
+        i {
+          display: none; // Hide icon
+        }
+      }
+    }
+  `}
+
+  // Apply special styling when boxyErrorStyle is true and the input has an error.
+  ${({ boxyErrorStyle, hasError }) =>
+    boxyErrorStyle &&
+    hasError &&
+    `box-sizing: content-box;
+    margin: calc(var(--spacing-8) * -1); // Negative margin to account for the padding, this is so the rest of the elements don't move
+    padding: var(--spacing-8);
+    border-radius: var(--border-radius);
+    background-color: var(--color-alertRedBg);`}
 `;
 
-interface InputWrapperProps {
-  className?: string;
-  children: JSX.Element | JSX.Element[];
+export interface InputWrapperProps {
   id: string;
-  alertLabel: string;
+  alert?: string;
+  children: JSX.Element | JSX.Element[];
```

Comment:
Change to React.ReactNode

Reply:
No can do.. It breaks the child clone logic, so this one is on purpose.

Reply:
💡